plugin/code_actions.py

```diff
@@ -323,14 +351,19 @@ def on_selection_modified_async(self) -> None:
                 lambda: self._stored_region == current_region, async_thread=True)
 
     def fire_request(self, current_region: sublime.Region) -> None:
-        self._actions = []
-        actions_manager.request(self.view, self.handle_responses, current_region.begin())
+        self._actions_by_config = {}
+        view = self.view
+        stored_range = Range(Point(*view.rowcol(self._stored_region.begin())),
+                             Point(*view.rowcol(self._stored_region.end())))
+        diagnostics_by_config = filter_by_range(view_diagnostics(view), stored_range)
+        actions_manager.request_with_optional_diagnostics(view, diagnostics_by_config, self.handle_responses)
 
     def handle_responses(self, responses: CodeActionsByConfigName) -> None:
-        for _, items in responses.items():
-            self._actions.extend(items)
-        if len(self._actions) > 0:
-            self.show_bulb()
+        self._actions_by_config = responses
+        if any((action for action in self._actions_by_config.values())):
+            self.show_annotations()
+            if settings.show_code_actions_bulb:
+                self.show_bulb()
```

Comment:
Shouldn't there be an `else` case here that removes bulbs and/or annotations in case the result params are empty?

Reply:
But those are removed when starting the request at `on_selection_modified_async`.